crates/forge_permission_resolver/src/permissions_cache.rs

```diff
@@ -0,0 +1,258 @@
+use regex::Regex;
+use std::collections::HashMap;
+use std::env;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use crate::permissions_resolver::PermissionHashMap;
+use crate::serde::{SerializablePermissionHashMap, ToStringMap};
+use std::fs::{self, File};
+use std::io::{Read, Write};
+use tracing::debug;
+
+const CACHE_EXPIRATION: Duration = Duration::from_secs(60 * 60 * 24 * 7); // 1 week
+
+fn default_cache_path() -> PathBuf {
+    let home = env::var("HOME").unwrap_or_else(|_| ".".to_string());
+    PathBuf::from(home).join(".cache/fsrt")
+}
+
+/// Struct to hold cache-related settings
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct CacheConfig {
+    use_cache: bool,
+    cache_path: Option<PathBuf>,
+    ttl: Duration,
+}
+
+impl CacheConfig {
+    pub fn new(use_cache: bool, cache_path: Option<PathBuf>) -> Self {
+        CacheConfig {
+            use_cache,
+            cache_path: Some(cache_path.unwrap_or_else(default_cache_path)),
+            ttl: CACHE_EXPIRATION,
+        }
+    }
+
+    pub fn use_cache(&self) -> bool {
+        self.use_cache
+    }
+
+    pub fn cache_path(&self) -> &PathBuf {
+        self.cache_path.as_ref().unwrap()
+    }
+}
+
+pub struct PermissionsCache {
+    config: CacheConfig,
+}
+
+impl PermissionsCache {
+    pub fn new(config: CacheConfig) -> Self {
+        PermissionsCache { config }
+    }
+
+    fn get_cache_path(&self, key: &str) -> PathBuf {
+        let mut path = self.config.cache_path().clone();
+        path.push(key);
+        path
+    }
+
+    fn is_cache_valid(&self, path: &PathBuf) -> bool {
+        if let Ok(metadata) = fs::metadata(path) {
+            if let Ok(modified) = metadata.modified() {
+                if let Ok(duration) = modified.elapsed() {
+                    if duration < self.config.ttl {
+                        return true;
+                    } else {
+                        println!(
```

Comment:
These should output to `stderr`, since it will clutter the output if someone redirects `fsrt`'s output to a file or program.